### PR TITLE
Added MultiValueCharField

### PR DESCRIPTION
**Pull request by HanlinMiao**
Added MultiValueCharField to accomodate the usage of MultiValueCharFilter that is not backed by choices.
# DEPENDENT ON
https://github.com/nautobot/nautobot/pull/2384
# TODO

- [x] Unit Tests
- [ ] Documentation
- [x] Make an `MultiValueCharField` for each `MultiValueCharFilter`

### CODE REVIEW

File: nautobot/utilities/forms/fields.py [hunk 1]
```diff
@@ -676,6 +676,25 @@ def has_changed(self, initial, data):
...
+        bound_field = BoundField(form, self, field_name)
+        value = bound_field.value()
+        widget = bound_field.field.widget
+        # Save the selected choices in the widget even after the filterform is submitted
+        if value is not None:
+            widget.choices = [(v, v) for v in value]
+
+        return bound_field
+
+
```
Comment (bryanculver) [inline nautobot/utilities/forms/fields.py:697]: Looks like this chain needs to be updated:

https://github.com/nautobot/nautobot/blob/develop/nautobot/utilities/filters.py#L527

https://github.com/nautobot/nautobot/blob/b027b6414c0be1222177793a8691793365354afa/nautobot/utilities/filters.py#L73-L74

https://github.com/nautobot/nautobot/blob/b027b6414c0be1222177793a8691793365354afa/nautobot/utilities/filters.py#L36-L60

Which should fix the magic generation issue, eliminating the need to specify the field.

And is the choices only helping on page reload pre-populate the inputted text? Is the necessary?

Comment (HanlinMiao) [reply nautobot/utilities/forms/fields.py:697]: Yeah the `widget.choices = [(v, v) for v in value] `are there after the filter is applied to keep the user input.

**Review by bryanculver — APPROVED** [review]
This should get merged and incorporated into #2348

**Review by glennmatthews — COMMENTED** [review]
Would it be desirable to add an integration test or two?

**Review by bryanculver — CHANGES_REQUESTED** [review]
> Would it be desirable to add an integration test or two?

Valid point @glennmatthews. I had more considered this being a component in the UI-side testing for #2348, however this does introduce a new Filter too which should be tested for the features it's providing.

**Comment by HanlinMiao** [comment]
Seems like `MultiValueCharFilter` is already being tested [here](https://github.com/nautobot/nautobot/blob/40e8db640611381b2d16e0feb75aefb4896daddb/nautobot/utilities/tests/test_filters.py#L512). Should we just test the `MultiValueCharField` by simulating how a user would use it in an browser(name, enter, name enter etc.) and see if it renders the correct result?

**Comment by bryanculver** [comment]
Merged via #2413

**Pull request closed without merging**